frame/staking-rewards/README.md

```diff
@@ -34,56 +33,74 @@ harvest `PICA` or `BTC` or `ETH` whenever a reward distribution occured.
-harvested rewards are still in possession of the user.
+Assuming the penalty is defined as being `0.5`, if I staked `10_000 PICA`,
+unstaking the NFT will result in a penalty of `5_000 PICA`. All harvested
+rewards are still in possession of the user.
 
-### Unstaking once the NFT expired
+### Unstaking when the NFT has no lock duration or if the lock expired
 
 In this canonical case, the user is able to unstake and rapatriate it's staked
 asset with no penalty.
 
 ### Claiming
 
-A user is able to claim his pending rewards at any point in time.
-As the protocol is rewarding the NFT itself, all pending rewards are tied to the NFT only.
-Meaning that if a user trade a NFT including pending rewards, the new owner will be able to unlock them.
+A user is able to claim his pending rewards at any point in time. As the
+protocol is rewarding the NFT itself, all pending rewards are tied to the NFT
+only. Meaning that if a user trade a NFT including pending rewards, the new
+owner will be able to unlock them.
 
 ### Rewarding stakers
 
-Any protocol is able to rewards the stakers by calling the
-`StakingRewards::transfer_reward` implementation of this pallet.
+Any protocol is able to rewards specific asset stakers by calling
+`StakingRewards::transfer_reward`. The protocol must provide the `asset` to
+reward (a.k.a. fNFT minted for staking `asset`) as well as the `reward_asset`
+along the `reward` amount. Depending on the pallet state, the reward will either
+be stored in the `PendingEpochRewards` or `EpochRewards` storage.
+
+### Epochs
 
-### Epoches
+Rewards are accumulated in epochs.
 
-Rewards accumulated in epochs.
+A user is able to stake an amount to mint a fNFT, but the registration of the
+NFT to get rewarded will happen at the next epoch (i.e. a newly minted fNFT
+start getting rewarded at the next epoch only).
 
-User may stake amount and get fNFT, but it will not start earn until next epoch start.
+After each epoch end, there is short period of time it takes to collect rewards
+and put new fNFTs into rewarding state. Once an epoch end, the pallet will
+advance in two states:
+- `Rewarding`: the pallet will reward all position by folding on the fNFT over
+  blocks, i.e. processing `X` distribution per block.
+- `Registering`: once the `Rewarding` state is done, the pallet register new
+stakers (a.k.a. fNFT created within the epoch that is currently ending). Once
+the `Registering` state is done, the pallet come back to a normal
+`WaitingForEpochEnd** state.
 
-After each epoch end, there is short period of time it takes to collect rewards and put new fNFTs into rewarding state.
+**NOTES**:
+- fNFTs are protected from dillution within an epoch as the dillution is
+  occuring in the above `Registering` state (a.k.a. registering new stakers).
+- users are able to interact with the pallet in the `WaitingForEpochEnd` state
+  only, any attempt while the pallet is `Registering` or `Rewarding` will result
+  in a `PalletIsBusy` error.
```

Comment:
I think name `PalletIsBusy` seems like a technical detail that user's won't care about. We can name it something like `BlockedForRewardsProcessing`

Reply:
A/B streams solves this nicely - it will never be busy.

Reply:
What are A/B streams?